### PR TITLE
feat(node): add dockerfile for building node container.

### PR DESCRIPTION
**Pull request by danwbyrne**
connects to https://quay.io/repository/neoone/node for automated building.

#706

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -416,6 +415,15 @@ const buildAll = ((cache) =>
     )(done);
   }))({});
 
+const buildAll = ((cache) =>
+  memoizeTask(cache, function buildAll(format, done, type) {
+    return gulp.parallel(
+      ...[buildAllNoDeveloperTools(format, done, type), compileDeveloperTools(format)].filter(
```
Comment (dicarlo2) [inline gulpfile.js]: Remove the `filter`

File: scripts/DockerfileNode [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+COPY packages/neo-one-client-full-core/package.json packages/neo-one-client-full-core/
+COPY packages/neo-one-client-full/src/ packages/neo-one-client-full/src
+COPY packages/neo-one-client-full/package.json packages/neo-one-client-full/
+COPY packages/neo-one-client-switch/src/ packages/neo-one-client-switch/src
+COPY packages/neo-one-client-switch/package.json packages/neo-one-client-switch/
+COPY packages/neo-one-node-bin/src/ packages/neo-one-node-bin/src
+COPY packages/neo-one-node-bin/package.json packages/neo-one-node-bin/
+COPY packages/neo-one-node-blockchain/src/ packages/neo-one-node-blockchain/src
+COPY packages/neo-one-node-blockchain/package.json packages/neo-one-node-blockchain/
+COPY packages/neo-one-node-browser-worker/src/ packages/neo-one-node-browser-worker/src
```
Comment (dicarlo2) [inline scripts/DockerfileNode]: Remove all of the `neo-one-node-browser` and `neo-one-worker`

File: scripts/DockerfileNode [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# Build Stage
+FROM neotracker/neo-one-circleci-node:10.13.0-1 as builder
+RUN mkdir -p tmp/neo-one
+WORKDIR tmp/neo-one
+
+COPY package.json tsconfig.json lerna.json gulpfile.js yarn.lock ./
+COPY scripts/ scripts
```
Comment (dicarlo2) [inline scripts/DockerfileNode]: do we need scripts?

File: scripts/DockerfileNode [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+COPY packages/neo-one-node-storage-levelup/package.json packages/neo-one-node-storage-levelup/
+COPY packages/neo-one-node-vm/src/ packages/neo-one-node-vm/src
+COPY packages/neo-one-node-vm/package.json packages/neo-one-node-vm/
+COPY packages/neo-one-node/src/ packages/neo-one-node/src
+COPY packages/neo-one-node/package.json packages/neo-one-node/
+
+RUN sudo yarn install --non-interactive
+RUN sudo yarn build:node
+
+#Final Stage
```
Comment (dicarlo2) [inline scripts/DockerfileNode]: Meh, remove the comments.

File: scripts/DockerfileNode [hunk 5]
```diff
@@ -0,0 +1,72 @@
+# Build Stage
```
Comment (dicarlo2) [inline scripts/DockerfileNode]: Remove

File: scripts/DockerfileNode [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+COPY packages/neo-one-node-storage-common/src/ packages/neo-one-node-storage-common/src
+COPY packages/neo-one-node-storage-common/package.json packages/neo-one-node-storage-common/
+COPY packages/neo-one-node-storage-levelup/src/ packages/neo-one-node-storage-levelup/src
+COPY packages/neo-one-node-storage-levelup/package.json packages/neo-one-node-storage-levelup/
+COPY packages/neo-one-node-vm/src/ packages/neo-one-node-vm/src
+COPY packages/neo-one-node-vm/package.json packages/neo-one-node-vm/
+COPY packages/neo-one-node/src/ packages/neo-one-node/src
+COPY packages/neo-one-node/package.json packages/neo-one-node/
+
+RUN sudo yarn install --non-interactive
```
Comment (dicarlo2) [inline scripts/DockerfileNode]: `sudo` is really necessary?

Comment (danwbyrne) [reply scripts/DockerfileNode]: if I don't sudo the `yarn install` I get:
```warning "react-static > eslint-config-react-tools > eslint > table > ajv-keywords@2.1.1" has incorrect peer dependency "ajv@^5.0.0".
error An unexpected error occurred: "EACCES: permission denied, mkdir '/tmp/neo-one/packages/neo-one-node-bin/node_modules'".```

I don't need to sudo the yarn build I can make that change.

File: scripts/DockerfileNode (hunk 3 above)
Comment (danwbyrne) [reply scripts/DockerfileNode]: no

File: gulpfile.js [hunk 7]
```diff
@@ -416,6 +415,11 @@ const buildAll = ((cache) =>
     )(done);
   }))({});
 
+const buildAll = ((cache) =>
+  memoizeTask(cache, function buildAll(format, done, type) {
+    return gulp.parallel(...[buildAllNoDeveloperTools(format, done, type), compileDeveloperTools(format)])(done);
```
Comment (dicarlo2) [inline gulpfile.js]: `buildAllNoDeveloperTools(format, type)`

**Comment by danwbyrne** [comment]
:thinking: @dicarlo2

**Pull request merged**